posthog/api/dashboard.py

```diff
@@ -187,7 +189,18 @@ def get_queryset(self) -> QuerySet:
     def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> response.Response:
         pk = kwargs["pk"]
         queryset = self.get_queryset()
-        dashboard = get_object_or_404(queryset, pk=pk)
+        try:
+            dashboard = get_object_or_404(queryset, pk=pk)
+        except Http404 as e:
+            user = cast(User, request.user)
+            alternative_team = Dashboard.objects.filter(deleted=False, team__in=user.teams)
```

Comment:
Aren't we missing the `pk` filter here?
```suggestion
            alternative_team = Dashboard.objects.filter(deleted=False, team__in=user.teams, pk=pk)
```

Reply:
🤦